src/Core/Grid/Column/ColumnCollection.php

```diff
@@ -107,14 +107,43 @@ public function toArray()
         return $columns;
     }
 
+    /**
+     * Move an existing Column to a specific position.
+     *
+     * @param string $id the Column ID original position in the Collection
+     * @param int $position the Column ID destination position in the Collection
+     *
+     * @return self
+     */
+    public function move($id, $position)
+    {
+        if (!isset($this->items[$id])) {
+            throw new ColumnNotFoundException(sprintf(
+                'Cannot insert new column into collection. Column with id "%s" was not found.',
+                $id
+            ));
+        }
+
+        $column = $this->items[$id];
+        unset($this->items[$id]);
+
+        $columns = array_slice($this->items, 0, $position - 1, true) +
```

Comment:
I don't think index position on 1 is a good idea. It is confusing for developers

Reply:
its fixed